### PR TITLE
rudimentary support for --stats in --dbt --json mode

### PR DESCRIPTION
**Pull request by stefankeidel**
We're working on a ci/cd implementation of data-diff for a dbt project in azure devops and a full diff is just taking too long.

This here change works for my particular case and I was wondering if maybe it's useful for the upstream project as well. 

I wasn't able to get the tests to run locally on a M1 mac, sadly. Would be fantastic if there was a dockerised setup one could run end-to-end for that

### CODE REVIEW

**Comment by dlawin** [comment]
@vvkh could you review this? Specifically the changes in format.py?

File: data_diff/format.py [hunk 1]
```diff
@@ -40,6 +40,7 @@ def jsonify(
     dataset2_columns: Columns,
     columns_diff: Dict[str, List[str]],
     with_summary: bool = False,
+    stats_only: bool = False,
```
Comment (vvkh) [inline data_diff/format.py:43]: Do you mind adding a simple test case [here](https://github.com/datafold/data-diff/blob/main/tests/test_format.py#L11)? At least one checking that `rows` is `None` when `stats_only=True`.

**Review by vvkh — CHANGES_REQUESTED** [review]
@stefankeidel thank you for contribution!
LGTM, let's add a simple test before merge.

File: data_diff/format.py (hunk 1 above)
Comment (dlawin) [reply data_diff/format.py:43]: > I wasn't able to get the tests to run locally on a M1 mac, sadly. Would be fantastic if there was a dockerised setup one could run end-to-end for that

Looks like they're having trouble running the tests. I can spend some time tomorrow adding one

**Comment by stefankeidel** [comment]
Apologies for not being particularly responsive, I’m currently on vacation. Should be back early next week. Thanks for taking the time to look at this!On 9. Aug 2023, at 09:41, Dan Lawin ***@***.***> wrote:﻿
@dlawin commented on this pull request.



In data_diff/format.py:
> @@ -40,6 +40,7 @@ def jsonify(
     dataset2_columns: Columns,
     columns_diff: Dict[str, List[str]],
     with_summary: bool = False,
+    stats_only: bool = False,


I wasn't able to get the tests to run locally on a M1 mac, sadly. Would be fantastic if there was a dockerised setup one could run end-to-end for that

Looks like they're having trouble running the tests. I can spend some time tomorrow adding one

File: data_diff/format.py (hunk 1 above)
Comment (dlawin) [reply data_diff/format.py:43]: added a test case

**Review by vvkh — APPROVED** [review]

File: data_diff/format.py [hunk 2]
```diff
@@ -53,21 +54,31 @@ def jsonify(
...
     diff_rows = []
+    rows = None
     schema = [field for field, _ in diff_info.diff_schema]
 
     t1_exclusive_rows, t2_exclusive_rows, diff_rows = _group_rows(diff_info, schema)
 
-    diff_rows_jsonified = []
-    for row in diff_rows:
-        diff_rows_jsonified.append(_jsonify_diff(row, key_columns))
+    if not stats_only:
```
[the remark this answers is not in the document]
Comment (stefankeidel) [reply data_diff/format.py:62]: Yep, makes sense. Just pushed a change implementing that :)

**Review by vvkh — APPROVED** [review]

**Review by vvkh — CHANGES_REQUESTED** [review]
hey @stefankeidel, looks like one of the tests is now failing, can you take a look?

**Comment by stefankeidel** [comment]
> hey @stefankeidel, looks like one of the tests is now failing, can you take a look?

Yep, sorry. I'm still unable to run the tests locally, which makes verifying without CI/CD a bit hard :). **Hopefully** good to go now, but let's see.

**Review by vvkh — APPROVED** [review]

**Comment by vvkh** [comment]
@stefankeidel LGTM now, appreciate the effort 🙇

**Pull request merged**